hail/src/main/scala/is/hail/io/RowStore.scala

```diff
@@ -986,343 +935,245 @@ object EmitPackDecoder {
   }
 }
 
-//
-// Generate the Type-specific C++ code for a PackDecoder
-//
-object NativeDecode {
-
-  def appendCode(sb: StringBuilder, rowType: PType, wantType: PType): Unit = {
-    val verbose = false
-    var seen = new ArrayBuffer[Int]()
-    val localDefs = new StringBuilder()
-    val entryCode = new StringBuilder()
-    val mainCode = new StringBuilder()
-
-    def stateVarType(name: String): String = {
-      name match {
-        case "len" => "ssize_t"
-        case "idx" => "ssize_t"
-        case "miss" => "std::vector<char>"
-        case _ => "char*"
-      }
+object NativeDecoder {
+  def skipBinary(input_buf_ptr: cxx.Expression): cxx.Code = {
+    val len = cxx.Variable("skip_len", "int", s"$input_buf_ptr->read_int()")
+    s"""${len.define}
+       |$input_buf_ptr->skip_bytes($len);""".stripMargin
```

Comment:
I have to admit, I don't really like this |.splitMargin stuff.  I find it just makes it noisier so I turned it off in IntelliJ and just put the generated text on the left, although then the indentation doesn't match the surrounding code.  I'm not sure which to prefer.  You don't need to change anything, I'm just thinking out loud.

Reply:
heh, I almost asked if you could change yours because I was having trouble parsing the structure. Do you think it would be reasonable to get rid of the stripMargin, but keep the indentation? That shouldn't affect the compilation, right?

And if we need to print the generated code for legibility, we can always shove it through `Pretty` which will do the right thing.